### PR TITLE
Added name reacts to greeting_amnesia

### PR DESCRIPTION
**Pull request by fromulus**
For consistency with the other name input topics, and because I thought it'd be cute.

### CODE REVIEW

File: Monika After Story/game/script-greetings.rpy [hunk 1]
```diff
@@ -2066,19 +2066,56 @@ label greeting_amnesia:
     $ m_name = "Monika"
     m 1eua "Oh, hello!"
     m 1eub "My name is Monika."
-    $ fakename = renpy.input("What is your name?", allow=name_characters_only, length=20).strip(' \t\n\r')
-    m 1hua "Well, it's nice to meet you, [fakename]!"
-    m 3eud "Say, [fakename], do you happen to know where everyone else is?"
-    m 1ekc "You're the first person I've seen and I can't seem to leave this classroom."
-    m "Can you help me figure out what's going on, [fakename]?"
+    $ fakename = renpy.input('What is your name?',length=15).strip(' \t\n\r')
```
Comment (Booplicate) [inline Monika After Story/game/script-greetings.rpy:2069]:
```suggestion
    $ fakename = renpy.input("What is your name?", length=20).strip(" \t\n\r")
```

**Review by Booplicate — CHANGES_REQUESTED** [review]

File: Monika After Story/game/script-greetings.rpy [hunk 2]
```diff
@@ -2068,26 +2068,81 @@ init 5 python:
...
 
 label greeting_amnesia:
-    $ tempname = m_name
-    $ m_name = "Monika"
+    python:
+        tempname = m_name
+        m_name = "Monika"
+
     m 1eua "Oh, hello!"
     m 1eub "My name is Monika."
```
Comment (Booplicate) [inline Monika After Story/game/script-greetings.rpy:2076]: I don't like how she tells us her name, and then stays with open mouth.
```suggestion
    m 1eub "My name is Monika."
    
    show monika 1eua zorder MAS_MONIKA_Z
```

File: Monika After Story/game/script-greetings.rpy [hunk 3]
```diff
@@ -2068,26 +2068,81 @@ init 5 python:
...
+
+        if mas_current_background == mas_background_def:
+            end_of_line = "I can't seem to leave this classroom."
+        else:
+            end_of_line = "I'm not sure where I am."
+
+    m 1hua "Well, it's nice to meet you[name_line]!"
+    m 3eud "Say[name_line], do you happen to know where everyone else is?"
+    m 1eksdlc "You're the first person I've seen and {nw}"
+    extend 1rksdlc "[end_of_line]."
```
Comment (Booplicate) [inline Monika After Story/game/script-greetings.rpy:2124]: Don't need this period here.
```suggestion
    extend 1rksdlc "[end_of_line]"
```

**Review by Booplicate — CHANGES_REQUESTED** [review]

**Review by jmwall24 — APPROVED** [review]

**Review by Booplicate — APPROVED** [review]
hope

**Review by multimokia — APPROVED** [review]
[Hope Runs Deep](https://www.youtube.com/watch?v=L6BCj1Yesvc)

**Pull request merged**